src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs

```diff
@@ -0,0 +1,123 @@
+
+		protected void PublishAllEntries(HealthReport report, Action<ServiceFabricHealth.HealthInformation> publishFunc,
+			 CancellationToken cancellationToken)
+		{
+			try
+			{
+				// We trust the framework to ensure that the report is not null and doesn't contain null entries.
+				foreach (KeyValuePair<string, HealthReportEntry> entryPair in report.Entries)
+				{
+					cancellationToken.ThrowIfCancellationRequested();
+					publishFunc(BuildSfHealthInformation(entryPair.Key, entryPair.Value));
+				}
+
+				cancellationToken.ThrowIfCancellationRequested();
+				publishFunc(BuildSfHealthInformation(report));
+			}
+			catch (FabricObjectClosedException)
+			{
+				// Ignore, the service instance is closing.
+			}
+		}
+
+
+		protected ServiceFabricHealth.HealthInformation BuildSfHealthInformation(HealthReport report)
+		{
+			int entriesCount = report.Entries.Count;
+			int healthyEntries = 0;
+			int degradedEntries = 0;
+			int unhealthyEntries = 0;
+			foreach (KeyValuePair<string, HealthReportEntry> entryPair in report.Entries)
+			{
+				switch (entryPair.Value.Status)
+				{
+					case HealthStatus.Healthy:
+						healthyEntries++;
+						break;
+					case HealthStatus.Degraded:
+						degradedEntries++;
+						break;
+					case HealthStatus.Unhealthy:
+					default:
+						unhealthyEntries++;
+						break;
+				}
+			}
+
+			StringBuilder descriptionBuilder = StringBuilderPool.Get();
+
+			descriptionBuilder
+				.AppendFormat("Health checks executed: {0}. ", entriesCount)
+				.AppendFormat("Healthy: {0}/{1}. ", healthyEntries, entriesCount)
+				.AppendFormat("Degraded: {0}/{1}. ", degradedEntries, entriesCount)
+				.AppendFormat("Unhealthy: {0}/{1}.", unhealthyEntries, entriesCount)
+				.AppendLine()
+				.AppendFormat("Total duration: {0}.", report.TotalDuration)
+				.AppendLine();
+
+			string description = descriptionBuilder.ToString();
+
+			ServiceFabricHealth.HealthInformation healthInfo = FinalizeHealthReport(HealthReportSummaryProperty, ToSfHealthState(report.Status));
+			healthInfo.Description = description;
+			return healthInfo;
+		}
+
+		protected ServiceFabricHealth.HealthState ToSfHealthState(HealthStatus healthStatus) =>
+			healthStatus switch
+			{
+				HealthStatus.Healthy => ServiceFabricHealth.HealthState.Ok,
+				HealthStatus.Degraded => ServiceFabricHealth.HealthState.Warning,
+				HealthStatus.Unhealthy => ServiceFabricHealth.HealthState.Error,
+				_ => throw new ArgumentException($"'{healthStatus}' is not a valid health status."),
+			};
+
+		protected ServiceFabricHealth.HealthInformation BuildSfHealthInformation(string healthCheckName, HealthReportEntry reportEntry)
+		{
+			StringBuilder descriptionBuilder = StringBuilderPool.Get();
+
+			if (!string.IsNullOrWhiteSpace(reportEntry.Description))
+			{
+				descriptionBuilder.Append("Description: ").Append(reportEntry.Description).AppendLine();
+			}
+			descriptionBuilder.Append("Duration: ").Append(reportEntry.Duration).Append('.').AppendLine();
```

Comment:
Would it make sense to also include the current correlation id/activity id here? Or is it expected to be part of the description?

Reply:
This part existed before - This PR convers only making rest health publisher and doesn't change old functionality.